### PR TITLE
eslint + prettier - jshint

### PR DESCRIPTION
**Pull request by umidbekkarimov**
See #742, #1096

### CODE REVIEW

**Comment by umidbekkarimov** [comment]
I didn't modify code to not create merge conflicts.

Preferred eslint rules marked as `warn`.

Run `npm run format && npm run lint:fix` to see the beauty.

File: pages/src/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "env": {
+    "browser": true
+  },
+  "parserOptions": {
+    "ecmaVersion": 6,
+    "sourceType": "script"
+  }
+}
```
Comment (leebyron) [inline pages/src/.eslintrc]: won't this file override the parent eslintrc and remove the custom rules you set up? If the only difference is adding `env.browser` then I think it's safe to merge in

**Comment by leebyron** [comment]
This is awesome, nice work

File: pages/src/.eslintrc (hunk 1 above)
Comment (umidbekkarimov) [inline pages/src/.eslintrc]: Of course it's safe 👍 check http://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy

**Comment by umidbekkarimov** [comment]
Do not forget to `yarn format` this after merge 😊

File: pages/src/.eslintrc (hunk 1 above)
Comment (leebyron) [inline pages/src/.eslintrc]: TIL!

**Comment by leebyron** [comment]
I'd rather not include all the warnings - from past usage I've found warnings essentially get ignored. I think the eslintrc should either turn things off or make them errors.

Some examples of rules that are broken pervasively in this codebase and are safe to turn off:

- vars-on-top
- no-var
- consistent-return
- no-param-reassign

**Comment by umidbekkarimov** [comment]
> Some examples of rules that are broken pervasively in this codebase and are safe to turn off:

Done.

> I think the eslintrc should either turn things off or make them errors

Agree, I kept "worthy" rules as a `warn`s for discussion.

**Comment by umidbekkarimov** [comment]
Closing this from https://github.com/facebook/immutable-js/commit/bb2fbf1749a8dffc3efee3fc0033ba3d2457d252

**Pull request closed without merging**